src/pages-and-resources/app-settings-modal/AppSettingsModal.jsx

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+import classNames from 'classnames';
+
+import { useDispatch } from 'react-redux';
+import {
+  Form, Button, TransitionReplace, ModalLayer, Hyperlink,
+} from '@edx/paragon';
+import FormSwitchGroup from '../../generic/FormSwitchGroup';
+import messages from './messages';
+
+import { enableDisableApp } from '../data/thunks';
+import { CoursePageShape } from '../pages/PageCard';
+import EnabledDisabledBadge from '../../generic/enabled-disabled-badge/EnabledDisabledBadge';
+import AppConfigFormDivider from '../discussions/app-config-form/apps/shared/AppConfigFormDivider';
+
+function AppSettingsModal({
+  intl, settings, title, children, onClose, onApply, enableAppLabel,
+  enableAppHelp, learnMoreURL, learnMoreText,
+}) {
+  const dispatch = useDispatch();
+  const learnMoreLink = (
+    <Hyperlink destination={learnMoreURL || 'https://www.google.com'} target="_blank">{learnMoreText}</Hyperlink>
+  );
+  return (
+    <ModalLayer
+      isOpen
+      closeText={intl.formatMessage(messages.cancel)}
+      dialogClassName="modal-dialog-centered modal-lg"
+      onClose={onClose}
+    >
+      <div
+        role="dialog"
+        aria-label={title}
+        className={classNames(
+          'bg-white',
+          'd-flex',
+          'flex-column',
+          'mw-xs',
+          'p-3',
+        )}
+      >
+        <h3>{title}</h3>
+        <AppConfigFormDivider />
+        <Form onSubmit={onApply}>
```

Comment:
why are not we using the Formik library for form management? by convention we are using Formik forms in codebase.

Reply:
This PR started before that became convention :-) 

I'll update this code to use that.

Reply:
Actually I think in this case Formik might not work as well since the the base form doesn't know about the form elements injected in the child component.

Reply:
We can use `Formik` in AppSettingsModal I see no reason why we can’t.

 The structure of form does not look good to me and the modal is not closing on submit.
The `type=submit` will also work if you place this button inside form closing tag. I think this needs to be revisited.